server/mergin/sync/files.py

```diff
@@ -0,0 +1,129 @@
+# Copyright (C) Lutra Consulting Limited
+#
+# SPDX-License-Identifier: AGPL-3.0-only OR LicenseRef-MerginMaps-Commercial
+import datetime
+import os
+from dataclasses import dataclass
+from typing import Optional, List
+from marshmallow import fields, EXCLUDE, pre_load, post_load
+from pathvalidate import sanitize_filename
+
+from .. import ma
+
+
+def mergin_secure_filename(filename: str) -> str:
+    """Generate secure filename for given file"""
+    filename = os.path.normpath(filename)
+    return os.path.join(
+        *[
+            sanitize_filename(path, replacement_text="_")
+            for path in filename.split(os.sep)
+        ]
+    )
+
+
+@dataclass
+class FileMeta:
+    """Base class for every file object"""
+
+    path: str
+    checksum: str
+    size: int
+
+    def is_valid_gpkg(self):
+        """Check if diff file is valid"""
+        return self.size != 0
+
+
+@dataclass
+class LocalFile(FileMeta):
+    """File stored on server"""
+
+    location: str
```

Comment:
Do we really  need location column?

it is combination of version and sanitized file name, no?

Reply:
At the end I decided to keep location in favor of version + sanitized as it is the information that matters, this attribute is also in old metadata and is older than `sanitized_name`. Moreover it allows to change the way we store the files in future, e.g. to have some random hashes only (which I'd prefer - to have flat structure and drop actual version folders completely).